src/test/integration/emailPassword.test.ts

```diff
@@ -59,4 +61,89 @@ describe('EmailPasswordProvider', () => {
         }
         throw Error();
     });
+    describe('Reset password', () => {
+        let resetToken: string;
+        test('User can reset password', async () => {
+            const authenticator = getResetPasswordAuthenticator();
+            const { token } = await authenticator.resetPassword(email);
+            expect(token).toBeTruthy();
+            expect(token.length).toBeGreaterThan(0);
+            resetToken = token;
+        });
+        test('Non-existing user cannot reset password', async () => {
+            const authenticator = getResetPasswordAuthenticator();
+            try {
+                await authenticator.resetPassword('non-existing@test.com');
+                throw new Error('Expected to throw');
+            } catch (error) {
+                expect(error.message).toMatch((ERROR_CODE.UserNotFound as any).message);
+            }
+        });
+        test('User can change password', async () => {
+            const password = 'myNewStrongPassword';
+            const authenticator = getChangePasswordAuthenticator();
+            await authenticator.changePassword(resetToken, password);
+            const credentials = await authenticator.signInWithEmailAndPassword(email, password);
+            expect(Object.keys(credentials).sort()).toStrictEqual(['credentials', 'user'].sort());
+        });
+        test('Non-existing user cannot change password', async () => {
+            const authenticator = getChangePasswordAuthenticator();
+            try {
+                const options = getChangePasswordOptions();
+                const token = await getResetToken(options, { email: 'bad@user.app' } as any);
+                await authenticator.changePassword(token, 'newPassword');
+                throw new Error('Expected to throw');
+            } catch (error) {
+                expect(error.message).toMatch((ERROR_CODE.UserNotFound as any).message);
```

Comment:
The same as above, I think we just reject with generic "token is not valid". Also I think it makes more sense to fail on the getResetToken? I find it strange we can issue token for user that does not exist. Is there any reason for it?

Reply:
Fixed - replaced with invalid token message